components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -1237,6 +1242,13 @@ void AdsServiceImpl::TriggerNewTabPageAdEvent(
     const std::string& placement_id,
     const std::string& creative_instance_id,
     const ads::mojom::NewTabPageAdEventType event_type) {
+  if (!IsEnabled()) {
+    // Only record to P3A if user does not have ads enabled,
+    // as required by NTP-SI P3A spec
+    p3a::RecordNewTabPageEvent(local_state_, p3a_service_, creative_instance_id,
+                               event_type);
```

Comment:
I will look at adding notifications so that we can decouple p3a logic from this function

Reply:
could you please elaborate?

Reply:
The single responsibility of this function is to trigger an ad event, other business logic should be decoupled

Reply:
Resolving as ads cannot use the lib